brightener.h

```diff
@@ -1,17 +1,34 @@
 #pragma once
 #include <cstdint>
+#include <vector>
+#include <memory>    // For std::unique_ptr
+#include <stdexcept> // For std::runtime_error
+#include <limits>    // For std::numeric_limits
 
 struct Image {
-	int rows;
-	int columns;
-	uint8_t pixels[1024 * 1024]; // max 1k x 1k image
+    int rows;
+    int columns;
+    std::unique_ptr<uint8_t[]> pixels; // Use unique_ptr for automatic memory management
+
+    Image(int r, int c);
+
+    // Move constructor
+    Image(Image&& other) noexcept;
+
+    // Move assignment operator
+    Image& operator=(Image&& other) noexcept;
+
+    // Delete copy constructor and copy assignment operator
+    Image(const Image&) = delete;
+    Image& operator=(const Image&) = delete;
 };
 
 class ImageBrightener {
 private:
-	Image m_inputImage;
+    Image& m_inputImage; // Reference to input image
 public:
-	ImageBrightener(Image& inputImage);
-	int BrightenWholeImage();
-	Image GetImage();
+    ImageBrightener(Image& inputImage);
+    int BrightenWholeImage();
+    Image& GetImage(); // Return a reference to the Image
```

Comment:
This will give back a reference to a private data member. This defeats the purpose of making `m_inputImage` as private.

Reply:
made it const so it will prevent any kind of modification attempt